server/plugin.go

```diff
@@ -73,35 +73,36 @@ func (p *Plugin) OnActivate() error {
 	if err != nil {
 		return errors.Wrap(err, "failed to reconfigure configurator on startup")
 	}
-	p.mm.Log.Debug("initialized config service")
-
-	accessKey := os.Getenv("APPS_INVOKE_AWS_ACCESS_KEY")
-	if accessKey == "" {
-		p.mm.Log.Warn("APPS_INVOKE_AWS_ACCESS_KEY is not set. AWS apps won't work.")
+	conf := p.conf.GetConfig()
+	mode := "Self-managed"
+	if conf.MattermostCloudMode {
+		mode = "Mattermost Cloud"
 	}
-	secretKey := os.Getenv("APPS_INVOKE_AWS_SECRET_KEY")
-	if secretKey == "" {
-		p.mm.Log.Warn("APPS_INVOKE_AWS_SECRET_KEY is not set. AWS apps won't work.")
+	if conf.DeveloperMode {
+		mode += ", Developer Mode"
 	}
-	p.aws, err = aws.MakeClient(accessKey, secretKey, &p.mm.Log)
+	p.mm.Log.Debug("initialized config service: " + mode)
+
+	p.aws, err = upaws.MakeClient(conf.AWSAccessKey, conf.AWSSecretKey, conf.AWSRegion, &p.mm.Log)
 	if err != nil {
 		return errors.Wrap(err, "failed to initialize AWS access")
 	}
-	p.mm.Log.Debug("initialized AWS Client")
+	p.mm.Log.Debug("initialized AWS Client",
+		"region", conf.AWSRegion, "bucket", conf.AWSS3Bucket,
+		"access", utils.LastN(conf.AWSAccessKey, 7), "secret", utils.LastN(conf.AWSSecretKey, 4))
 
 	p.httpOut = httpout.NewService(p.conf)
 	p.mm.Log.Debug("initialized outgoing HTTP")
 
 	p.store = store.NewService(p.mm, p.conf)
 	// manifest store
-	conf := p.conf.GetConfig()
 	mstore := p.store.Manifest
 	mstore.Configure(conf)
-	// TODO: uses the default bucket name, do we need it customizeable?
-	manifestBucket := apps.S3BucketNameWithDefaults("")
-	err = mstore.InitGlobal(p.aws, manifestBucket, p.httpOut)
-	if err != nil {
-		return errors.Wrap(err, "failed to initialize the global manifest list from marketplace")
+	if conf.MattermostCloudMode {
```

Comment:
Does it make sense to also do that in dev mode given that the install command for AWS is available?